### ISSUE TITLE
`avc:  denied  { shutdown }` when using socket activation with rootless podman quadlet

### ISSUE THREAD

**Issue by gfieni** [containers/container-selinux#263]
Hello,

I am using Podman quadlet to manage containerized services under a non-root user.
I followed the [freedesktop.org tutorial](https://freedesktop.org/wiki/Software/systemd/DaemonSocketActivation/) and the [Podman documentation](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md) to have a socket activation for an Nginx container.
But when running the containers with `setenforce 1`, the following errors appear in Nginx logs for every connection :
```
2xxx:xxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx - - [11/Aug/2023:17:20:38 +0000] "GET / HTTP/2.0" 200 615 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36" "-"

[...]

2023/08/11 17:21:43 [alert] 4#4: *6 shutdown() failed (13: Permission denied) while processing HTTP/2 connection, client: 2xxx:xxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx, server: [::]:443
```

And there is a corresponding AVC denied error in the system logs :
```
Aug 11 17:21:43 test-selinux audit[3449]: AVC avc:  denied  { shutdown } for  pid=3449 comm="nginx" laddr=2yyy:yyy:yyyy:yyyy::y lport=443 faddr=2xxx:xxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx fport=54751 scontext=system_u:system_r:container_t:s0:c842,c880 tcontext=unconfined_u:un>
```

This problem does not appear for HTTP connections, only the HTTPS ones are affected.
I tried with `setenforce 0` and this problem disappears.
When running the Nginx container without the socket activation (using the `PublishPort=[::]:443:443/tcp` parameter in quadlet) there are also no problems.

### The systemd units :

`~/.config/systemd/user/test-nginx.socket`
```
[Unit]
Description=NGINX server socket
Before=test-nginx.service

[Socket]
ListenStream=[::]:443

[Install]
WantedBy=sockets.target
```

`~/.config/containers/systemd/test-nginx.container`
```
[Unit]
Description=Test NGINX server
Wants=test-nginx.socket

[Container]
Image=docker.io/library/nginx:mainline

Tmpfs=/docker-entrypoint.d:ro,size=1K,noexec,nosuid,nodev,notmpcopyup

Volume=%h/nginx/conf.d:/etc/nginx/conf.d:Z,ro
Volume=%h/nginx/ssl:/etc/nginx/ssl:Z,ro

# Use the systemd socket.
Environment=NGINX=3;

[Install]
WantedBy=default.target
```

`~/nginx/conf.d/10-https.conf`
```
server {
  listen [::]:443 ssl;
  http2 on;

  ssl_certificate /etc/nginx/ssl/cert.pem;
  ssl_certificate_key /etc/nginx/ssl/key.pem;
  ssl_protocols TLSv1.2 TLSv1.3;

  location / {
          root /usr/share/nginx/html;
  }

}
```

### About the machine :

```console
$ cat /etc/fedora-release
Fedora release 38 (Thirty Eight)
$ uname -a
Linux test-selinux 6.4.9-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Aug  8 21:21:11 UTC 2023 x86_64 GNU/Linux
$ rpm -q container-selinux podman systemd
container-selinux-2.219.0-1.fc38.noarch
podman-4.6.0-1.fc38.x86_64
systemd-253.7-1.fc38.x86_64
```

**Comment by rhatdan** [comment]
Could you paste the entire AVC message.

**Comment by gfieni** [comment]
Yes of course, sorry it was truncated :
```
Aug 11 17:21:43 test-selinux audit[3449]: AVC avc:  denied  { shutdown } for  pid=3449 comm="nginx" laddr=2yyy:yyy:yyyy:yyyy::y lport=443 faddr=2xxx:xxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx fport=54751 scontext=system_u:system_r:container_t:s0:c842,c880 tcontext=unconfined_u:unconfined_r:container_runtime_t:s0-s0:c0.c1023 tclass=tcp_socket permissive=0
```

**Comment by rhatdan** [comment]
Fixed in container-selinux v2.221.0
https://github.com/containers/container-selinux/releases/tag/v2.221
